workplace/views.py

```diff
@@ -513,11 +513,21 @@ def destroy(self, request, *args, **kwargs):
         empty response as if it was deleted, but will instead modify specific
         fields to keep a track of events. Subsequent delete request won't do
         anything, but will return a success.
+        An admin can also delete a reservation, but will have the choice to
+        give back the reservation ticket to the user
         """
         instance = self.get_object()
+        user = instance.user
         if instance.is_active:
             instance.is_active = False
-            instance.cancelation_reason = 'U'
             instance.cancelation_date = timezone.now()
+            if self.request.user.id != user.id:
+                instance.cancelation_reason = 'A'
+                ticket_return = request.data.get('ticket_return', False)
+                if ticket_return:
+                    user.tickets += 1
+                    user.save()
+            else:
+                instance.cancelation_reason = 'U'
```

Comment:
If an admin (ie: bob) try to cancel his own reservation with a refund. It will cancel the reservation WITHOUT the refund and complete the task. So Bob will lose money and will not have any option to correct the error.

The logic is not good.

Reply:
Corrected and test added